compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -1299,16 +1304,30 @@ private BRecordTypeSymbol getFrameTypeSymbol() {
     @Override
     public void visit(BLangLambdaFunction lambda) {
         BLangFunction function = lambda.function;
-        currentFrameSymbol = function.requiredParams.get(0).symbol;
-        identifiers = new HashMap<>();
-        currentLambdaBody = (BLangBlockFunctionBody) function.getBody();
-        List<BLangStatement> stmts = new ArrayList<>(currentLambdaBody.getStatements());
-        stmts.forEach(stmt -> {
-            stmt.accept(this);
-        });
-        currentFrameSymbol = null;
-        identifiers = null;
-        currentLambdaBody = null;
+        if (function.flagSet.contains(Flag.QUERY_LAMBDA)) {
+            currentFrameSymbol = function.requiredParams.get(0).symbol;
+            identifiers = new HashMap<>();
+            currentLambdaBody = (BLangBlockFunctionBody) function.getBody();
+            currentLambdaEnv = SymbolEnv.createFuncBodyEnv(currentLambdaBody, env);
+            List<BLangStatement> stmts = new ArrayList<>(currentLambdaBody.getStatements());
+            stmts.forEach(stmt -> stmt.accept(this));
+            currentFrameSymbol = null;
+            identifiers = null;
+            currentLambdaBody = null;
```

Comment:
Can you try out a nested lambda function example? Probably you'll have to maintain previous state and set at the end